rust/worker/src/execution/operators/hnsw_knn.rs

```diff
@@ -12,6 +13,8 @@ pub struct HnswKnnOperatorInput {
     pub segment: Box<DistributedHNSWSegmentReader>,
     pub query: Vec<f32>,
     pub k: usize,
+    pub allowed_ids: Arc<[String]>,
```

Comment:
do this really need to be Arc?

Reply:
Yes. Since the the BF KNN operator and its associated data will be wrapped in a task.